transform/variant/maf_transform.py

```diff
@@ -0,0 +1,226 @@
+        futures = (executor.submit(func, line) for line in lines)
+        for future in as_completed(futures):
+            yield future.result()
+
+
+def _read_maf(mafpath, gz):
+    """ generator for each line in maf """
+    if gz or 'gz' in mafpath:
+        inhandle = gzip.open(mafpath, mode='rt')
+    else:
+        inhandle = open(mafpath)
+    reader = csv.DictReader(inhandle, delimiter="\t")
+    for line in reader:
+        yield line
+    inhandle.close()
+
+
+class MafConverter():
+
+    def convert(self, emit, mafpath, source, genome='GRCh37',
+                method='variant', gz=False, centerCol='Center'):
+        """
+        emit -  a way to write output
+        mafpath - a file to read
+        source - special handling if 'tcga'
+        genome - reference_genome e.g. GRCh37
+        method - call's method
+        gz - is mafpath a gz
+        """
+        # center = 2
+        # ncbi_build = 3
+        chromosome = "Chromosome"  # 4
+        start = ["Start_Position", "Start_position"]  # 5
+        end = ["End_Position", "End_position"]  # 6
+        # strand = 7
+        variant_type = "Variant_Type"  # 9
+        reference_allele = "Reference_Allele"  # 10
+        # tumor_allele1 = "Tumor_Seq_Allele1"  # 11
+        tumor_allele2 = "Tumor_Seq_Allele2"  # 12
+        # annotation_transcript = "Annotation_Transcript"  # 14
+
+        tumor_sample_barcode = "Tumor_Sample_Barcode"  # 15
+        normal_sample_barcode = "Matched_Norm_Sample_Barcode"  # 16
+
+        # normal_allele1 = 17
+        # normal_allele2 = 18
+        # verification_status = 23
+        # validation_status = 24
+        # mutation_status = 25
+        # sequencing_phase = 26
+        # sequence_source = 27
+        # bam_file = 30
+
+        # sequencer = 31
+        # genome_change = 32
+
+        # # Information indices for VariantCallEffect and Gene
+        # entrez_gene_id = 1
+        # variant_classification = 8
+
+        feature = "Feature"  # 48
+        feature_type = "Feature_type"  # 49
+        dbSNP_RS = "dbSNP_RS"  # 13
+
+        logging.info('converting maf: ' + mafpath)
+
+        sample_ids = {}
+
+        def _allele_call_maker(allele, callset, line=None):
+            """ create call from line """
+            keys = ['t_depth', 't_ref_count', 't_alt_count', 'n_depth',
+                    'n_ref_count', 'n_alt_count']
+            info = {}
+            for k in keys:
+                info[k] = get_value(line, k, None)
+            return AlleleCall(allele.gid, callset.gid, info)
+
+        def _callset_maker(allele, line):
+            """ create callset from line """
+            barcode = line[tumor_sample_barcode]
+            sample = barcode
+            if source == 'tcga':
+                if barcode not in sample_ids:
+                    sample_ids[barcode] = _tcga_aliquot2sample_barcode(barcode)
+                sample = sample_ids[barcode]
+            sample = Biosample.make_gid(sample)
```

Comment:
The "sample" name is being used for two different types (barcode vs GID) which is confusing.

Reply:
agreed